packages/hash/api/src/graphql/resolvers/index.ts

```diff
@@ -221,6 +223,8 @@ export const resolvers = {
     resolvePersistedComment: loggedInAndSignedUp(resolvePersistedComment),
     deletePersistedComment: loggedInAndSignedUp(deletePersistedComment),
     updatePersistedCommentText: loggedInAndSignedUp(updatePersistedCommentText),
+    // HASH instance admin mutations
+    createUser: loggedInAndSignedUpHashInstanceAdmin(createUser),
```

Comment:
I am not suggesting we change anything here right now, but to me this feels like something that could be solved with "hooks" as well, but on entity creation. I appreciate we haven't fledged out all the use-cases and the authorization mechanisms for these hooks, but I imagine a "CRUD hook definition" would include pre/post checks that can handle validation/authorization/side effects. 

But as we've talked about previously, these hooks may even have to be closer to the Graph so other services using the Graph can benefit from them. Just a thought :)

Reply:
Yes, I did consider this as an option, but there would be a lot of complexities that I don't are worth solving right now, some of which enclude:
- the hook would need to create the `accountId`, and populate the `entityId` of the entity
- the hook would need to create a kratos identity, and populate the `kratosIdentityId` property on the entity
- the `createEntity` request would need to accept a "partial" `properties` object, as some of the properties (such as the `kratosIdentitiyId`) would need to be created in the hook. This also makes the method a lot harder to document/call correctly (how does the caller know which required `User` properties need to be supplied/not supplied, because they are/aren't created in the hooks?)

Reply:
cc @Alfred-Mountfield as he may also be interested in my reasoning